### DISCUSSION TITLE
Unable to download though @app.post using StreamingResponse

### DISCUSSION THREAD

**Discussion by skk294** [fastapi/fastapi#6538] [Questions] [answered]
### First Check

- [x] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI()

origins = ["*"]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)

@app.post("/download_test")
async def admin_download_domain_file(request: Request):
    import xlsxwriter
    from fastapi.responses import StreamingResponse, FileResponse
    from io import BytesIO

    output = BytesIO()
    workbook = xlsxwriter.Workbook(output)
    worksheet = workbook.add_worksheet()
    worksheet.write(0, 0, "test")
    workbook.close()
    headers = {"Content-Disposition": "filename=filename.xlsx"}
    return StreamingResponse(output, headers=headers)
```

### Description

I can't download file though @app.post, but if i change it to @app.get, it works fine.
The reason I need to use @app.post is because I need to send a list to server, maybe hundreds of url in it then return a xlsx file which contains their imformation.
I got these
--Failed to fetch.
--Possible Reasons:
--CORS
--Network Failure
--URL scheme must be "http" or "https" for CORS request.
in http://127.0.0.1:8000/docs
And everytime I use  POST method it will jump to GET method like this:
INFO:     127.0.0.1:44548 - "POST /download_test HTTP/1.1" 200 OK
INFO:     127.0.0.1:44688 - "GET /download_test HTTP/1.1" 405 Method Not Allowed
But I didn't define GET method. I have set allow_origins=['*'], it does't work.

### Operating System

Linux

### FastAPI Version

0.66.1

### Python Version

3.9.5

**Comment by Dustyposa** [comment]
I tried you code, but work fine. Like this:
```bash
INFO:     127.0.0.1:50412 - "GET /docs HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "GET /openapi.json HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "POST /download_test HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "POST /download_test HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "POST /download_test HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "POST /download_test HTTP/1.1" 200 OK
INFO:     127.0.0.1:50412 - "POST /download_test HTTP/1.1" 200 OK
```

**Comment by skk294** [comment]
@Dustyposa Thanks for your help. According to your words, I finally realize it's the XDM Browser Monitor's problem, it somehow make me jump to GET method. After shutting it down, everything works fine.

**Comment by sciai-ai** [comment]
Hi, I am getting the same problem at my end. I am returning a FileResponse but the get function gets called instead of rendering the audio/wav response. I dont define nay get function sa i need to return a Filereponse after an input json get posted. I belivee this is related to SwaggerUI perhaps [https://github.com/swagger-api/swagger-ui/pull/3441](url) 

![image](https://user-images.githubusercontent.com/52277510/130393099-bfc0e202-adc9-4276-bd6a-c694f140cacb.png)

**Comment by skk294** [comment]
> Hi, I am getting the same problem at my end. I am returning a FileResponse but the get function gets called instead of rendering the audio/wav response. I dont define nay get function sa i need to return a Filereponse after an input json get posted. I belivee this is related to SwaggerUI perhaps [https://github.com/swagger-api/swagger-ui/pull/3441](url)
> 
> ![image](https://user-images.githubusercontent.com/52277510/130393099-bfc0e202-adc9-4276-bd6a-c694f140cacb.png)

```
from fastapi import FastAPI
from fastapi.responses import FileResponse

file_path = "file_example_WAV_1MG.wav"
app = FastAPI()


@app.post("/download_wav")
async def download_wav():
    return FileResponse(file_path)
```
use `curl -X 'POST'   'http://127.0.0.1:8000/download_wav' --output file_example_WAV_1MG.wav` to test, its okay

**Comment by sciai-ai** [comment]
Hi, calling curl in shell works fine, its an issue with swagger UI which according to below references has been fixed. Can you please check which swagger UI version is integrated with fastapi current release. I could not get any info on that or how to access the swagger UI code in fastapi

[https://stackoverflow.com/questions/45224263/swagger-ui-does-not-display-image-png-content-in-post](url)

[https://github.com/swagger-api/swagger-ui/issues/3435](url)

[https://github.com/swagger-api/swagger-ui/pull/3441](url)

**Comment by jramirez92** [comment]
I'm having the same issue, Is there any way to update the Swagger version in FastApi?

**Comment by YuriiMotov** [comment, marked as answer]
Duplicate: https://github.com/fastapi/fastapi/discussions/6284
